internal/model/target.go

```diff
@@ -0,0 +1,29 @@
+package model
+
+import "fmt"
+
+// An abstract build graph of targets and their dependencies.
+// Each target should have a unique ID.
+
+type TargetType string
+
+const (
+	TargetTypeK8s           TargetType = "k8s"            // Deployed k8s entities
+	TargetTypeImage         TargetType = "image"          // Image builds
+	TargetTypeDockerCompose TargetType = "docker-compose" // Docker-compose service build+deploy
```

Comment:
worth including in the note that we treat it as a deploy target right now but that that's sort of a hack

Reply:
done in the next pr